### PR TITLE
Main changes: created cherrypy server, improved security of monhelper, updated code to django 1.10

### PR DESCRIPTION
**Pull request by lciti**
I found suponoff an excellent dashboard for supervisor.
I have implemented some extra features I needed and I propose their inclusion in the main branch.
In particular:
- I have created a standalone server for the suponoff app using the production-ready cherrypy server
- I have made some changes to monhelper so it can be used with enhanced security
- I have made the code compatible with versions of django from 1.7 to 1.10

### CODE REVIEW

File: suponoff-monhelper.py [hunk 1]
```diff
@@ -167,15 +175,26 @@ def split_namespec(namespec):
     return group_name, process_name
 
 
-def main():
+def main(args=None):
+    import argparse
```
Comment (gjcarneiro) [inline suponoff-monhelper.py]: Import inside function not justified, needs to be moved to toplevel.

File: suponoff-server.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        ]
+    )
+
+
+def main(args=None):
+
+    import cherrypy
+    from django.core.handlers.wsgi import WSGIHandler
+    import argparse
+
```
Comment (gjcarneiro) [inline suponoff-server.py]: Again more imports inside functions is not pythonic.

File: suponoff/views.py [hunk 3]
```diff
@@ -8,7 +8,10 @@
 import xmlrpc.client
 
 import concurrent.futures
-from django.core.context_processors import csrf
+try: # django>=1.8
+    from django.template.context_processors import csrf
+except: # django==1.7
```
Comment (gjcarneiro) [inline suponoff/views.py]: `except ImportError`, please.  Too wide exception catch.

File: suponoff-monhelper.py [hunk 4]
```diff
@@ -42,10 +42,15 @@ def getProcessResourceUsage(self, pid_list):
...
             results = {}
             for pid in pid_list:
+                if pid == 0:
+                    continue
                 result = {}
                 results[str(pid)] = result
                 try:
                     proc = psutil.Process(pid)
+                except psutil.NoSuchProcess as e:
+                    LOG.info("Process %s: %s", pid, e.msg)
```
Comment (gjcarneiro) [inline suponoff-monhelper.py]: Perhaps here it should log "No such process" instead of "Process"?

**Review by gjcarneiro — CHANGES_REQUESTED** [review]
Thanks for the PR.  I don't have any major objection, just some minor code style changes requested.

File: suponoff-monhelper.py (hunk 4 above)
Comment (lciti) [reply suponoff-monhelper.py]: I am happy either way. Please notice that e.msg already describes the error, "Process %s: %s" gives: "Process 17487: no process found with pid 17487".

File: suponoff/views.py (hunk 3 above)
Comment (lciti) [reply suponoff/views.py]: Sure.

File: suponoff-monhelper.py (hunk 1 above)
Comment (lciti) [reply suponoff-monhelper.py]: I usually try to stick to PEP 8. My only exception is (like in this case) when a file could be used as a module and as a main. In this case I leave the main-specific imports within main. In any case, I am happy to move these imports to the beginning of the file as you prefer.

File: suponoff-server.py (hunk 2 above)
Comment (lciti) [reply suponoff-server.py]: Ok here too

**Review by gjcarneiro — APPROVED** [review]
Looks good, thanks!

**Comment by lciti** [comment]
Great, many thanks to you!

**Pull request merged**